docs/DesignDocs/CustomAttestation.md

```diff
@@ -0,0 +1,420 @@
+___
+
+Custom Attestation Data Formats for Open Enclave
+=====
+
+This design document proposes a new attestation framework and set of APIs that
+enable developers to use custom formats for their attestation evidence and
+collateral.
+
+Motivation
+----------
+
+Currently, Open Enclave provides several APIs that developers can use for
+attestation. The two key functions are `oe_get_report`, which produces an opaque
+blob that is signed by the enclave, and `oe_verify_report`, which can be used to
+verify the generated report. The original purpose of those two APIs were to
+provide a simple, cross-platform way to produce and verify attestation data.
+
+However, for some developers, these APIs are too inflexible for their
+attestation requirements. For example, one might want to extend Open Enclave's
+current attestation structures with extra information, such as geolocation
+or a timestamp. Another user might want their enclaves to generate attestation
+data that is in a compatible format with their existing authentication
+infrastructure, such as a JSON Web Token or a X509 certificate. Overall,
+there has been interest in enchancing Open Enclave's APIs to support custom
+attestation formats to enable these scenarios.
+
+Specification
+-------------
+
+To support custom attestation formats, this document proposes adding a plugin
+model for attestaton. The Open Enclave SDK will define a set of common APIs
+that each plugin must implement. Each plugin will define a random UUID to
+distinguish it from other plugins.
+
+Futhermore, there will be additional attestation "plugin aware" APIs that are
+analogous to `oe_get_report` and `oe_verify_report`, along with functions for
+registering and unregistering plugins. The user can statically link in their
+desired plugin and call the register plugin function. The attestation data can
+be retrieved from the "plugin aware" analogue of `oe_get_report` with the
+desired UUID. The generated data will have the UUID in its header. The user can
+call the analogue of `oe_verify_report` to verify the data and the Open Enclave
+runtime can use this UUID to determine what plugin verification routine to run.
+
```

Comment:
I don;t see anywhere the default SGX plug-ins got mentioned, which will remove confusion on what this plugin mode is supporting

Reply:
I added a section for the SGX plugin.